workorders.lic

```diff
@@ -260,15 +260,22 @@ class WorkOrders
   def shape_items(info, item, quantity)
     ensure_copper_on_hand(5000, @hometown)
     recipe, items_per_stock, spare_stock, scrap = find_recipe(info, item, quantity)
+    stock_name = 'stock-name'
+    stock_number = 'stock-number'
 
     quantity.times do |count|
-      if (count % items_per_stock).zero?
+      if items_per_stock.zero? || (count % items_per_stock).zero?
         dispose("#{info['stock-name']} lumber") if count > 0 && spare_stock
         if count > 0
           go_door
           pause 0.5 until Room.current.id
         end
-        order_item(info['stock-room'], info['stock-number'])
+        fput("get my #{info[stock_name]} lumber")
```

Comment:
Could you use `bput` here?

Reply:
for some reason switching to bput("get my #{info[stock_name]} lumber", 'What were', 'You get') kills it, after the first item it crashes with 

> --- Lich: shape has exited.
> --- Lich: error: undefined local variable or method `stock' for #<Scripting::WorkOrders:0x2f3a990>
> 	test-wo:268:in `block in shape_items'
> 	test-wo:266:in `times'

Reply:
That's a very unexpected error, could you post the modified code +/- maybe 20 lines of code?

Reply:
Scratch that, it was actually this change that broke it,  not the bput.

-        dispose("#{info['stock-name']} lumber") if count > 0 && spare_stock
+        dispose("#{info[stock-name]} lumber") if count > 0 && spare_stock

Reply:
That would need to be `info[stock_name]` (note the underscore) to work.